qiskit/transpiler/synthesis/permrowcol.py

```diff
@@ -242,3 +242,24 @@ def _get_nodes_for_eliminate_row(
         ]  # Finds indexes of rows that are added to chosen_row
 
         return nodes
+
+    def _add_cnot(self, edge: tuple, circuit: QuantumCircuit):
+
+        """check if cnot is in allowed direction and add Hadamard gates if necessary
+
+        Args:
+            edge (tuple): tuple representing qubits between cnot.
+            circuit (QuantumCircuit): QuantumCircuit for adding cnots
+
+        """
+        edges = self._coupling_map.get_edges()
+
+        if edge not in edges:
+            circuit.h(edge[0])
+            circuit.h(edge[1])
+            circuit.cx(edge[0], edge[1])
```

Comment:
You do need to change the CNOT direction wrt the edge.

Reply:
So the CNOTs added to the circuit are supposed to be always in the right direction?